code/game/objects/items/stunbaton.dm

```diff
@@ -173,9 +172,32 @@
 		var/mob/living/carbon/human/H = L
 		H.forcesay(GLOB.hit_appends)
 
-
 	return 1
 
+/// After a target is hit, we do a chunk of stamina damage, along with other effects.
+/// After a period of time, we then check to see what stun duration we give.
+/obj/item/melee/baton/proc/stun_effect(mob/living/target)
+	target.Jitter(20)
+	target.confused = 10
+	target.apply_effect(EFFECT_STUTTER, stunforce)
+	target.adjustStaminaLoss(65)
+
+	SEND_SIGNAL(target, COMSIG_LIVING_MINOR_SHOCK)
+	addtimer(CALLBACK(src, .proc/apply_stun_effect_end, target), 20)
+
+/// After the initial stun period, we check to see if the target needs to have the stun applied.
+/obj/item/melee/baton/proc/apply_stun_effect_end(mob/living/target)

```

Comment:
```DM
/obj/item/melee/baton/proc/apply_stun_effect_end(mob/living/target)
    if(target.Paralyze)
        return
    if(HAS_TRAIT)
         to_chat
         paralyze
    else
         to_chat
         paralyze but different calc
```

Reply:
>  if(target.IsParalyzed())
>         return

This prevents further stuns when already paralyzed, definitely not an intention.

Reply:
don't resolve my comments if you didn't change anything please.

Reply:
You've changed what you said completely. Resolving was perfectly reasonable there... You can always just comment again to unresolve it lol.

Reply:
On second look this still isn't what I'm looking for - the logic is different. Is there an issue with what I currently have? I'm unsure if you're trying to actually change the logic of it, or refactor the code to be neater.

Reply:
This will always stun after the delay, ignoring my logic I currently have for the chemical shortening the stun (the trait check is inside the "if already stunned").

Reply:
How about this logic?

```
/obj/item/melee/baton/proc/apply_stun_effect_end(mob/living/target, variable/stunlength)
     variable tempvar = 1
     if(HAS_TRAIT)
          to_chat
          tempvar*=0.1
          return
     paralyze for a duration equal to tempvar*stunlength
```

Reply:
Now my issue with the stun message appearing when you’re already stunned is back since the !isparalyzed checks aren’t in. Plus you’re aren’t stunned at all now when you have the trait since you early return. 

Is there an issue with what I currently have, I don’t quite understand.

Reply:
third time is a charm ;)

Reply:
changed to this